src/components/icons/AnimatedIcon.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react'
+import { css, cx } from '../../../styled-system/css'
+import { icon } from '../../../styled-system/recipes'
+import { token } from '../../../styled-system/tokens'
+import { AnimatedIconType } from '../../@types/Icon'
+import { ICON_SCALING_FACTOR } from '../../constants'
+import LottieAnimation from './LottieAnimation'
+
+/** Animated Icon with Conditional Lottie Animation. */
+const AnimatedIcon = ({
+  fill,
+  size = 18,
+  style = {},
+  cssRaw,
+  animated,
+  animationData,
+  children,
+  animationComplete,
+}: AnimatedIconType) => {
+  const newSize = size * ICON_SCALING_FACTOR
+  const color = style.fill || fill || token('colors.fg')
+
+  // Local state to manage animation status
+  const [isAnimated, setIsAnimated] = useState(animated)
+
+  useEffect(() => {
+    setIsAnimated(animated)
+  }, [animated])
+
+  /** Calls the animationComplete callback if provided. */
+  const handleAnimationComplete = () => {
+    if (animationComplete) {
+      animationComplete()
+    }
+  }
```

Comment:
Can you pass `animationComplete` directly to `<LottieAnimation>` instead of having the wrapper?

Reply:
That’s a good point; we can pass animationComplete directly to <LottieAnimation> for now. I initially set it up with a wrapper to accommodate future enhancements, like ensuring a smooth reappearance of the animation.